knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java

```diff
@@ -77,19 +77,20 @@ public void doService() {
 			} else if (OPER_DOWNLOAD.equalsIgnoreCase(operation)) {
 				freezeHttpResponse();
 				String fileName = (String) getAttribute("fileName");
+
 				File file = getFile(fileName);
-				try (FileInputStream fis = new FileInputStream(file)) {
-					HttpServletResponse response = getHttpResponse();
-					response.setHeader("Content-Disposition", "attachment; filename=\"" + file.getName() + "\";");
-					byte[] content = SpagoBIUtilities.getByteArrayFromInputStream(fis);
-					response.setContentLength(content.length);
-					response.getOutputStream().write(content);
-					response.getOutputStream().flush();
-				} catch (Throwable t) {
-					logger.error("Error getting file", t);
+				if (!file.exists()) {
 					writeBackToClient(404, "Error getting file with name, not found", false, null, "text/plain");
 				}
 
+				FileInputStream fis = new FileInputStream(file);
+				HttpServletResponse response = getHttpResponse();
+				response.setHeader("Content-Disposition", "attachment; filename=\"" + file.getName() + "\";");
+				byte[] content = SpagoBIUtilities.getByteArrayFromInputStream(fis);
+				response.setContentLength(content.length);
+				response.getOutputStream().write(content);
+				response.getOutputStream().flush();
+
```

Comment:
this part must be inserted within an else block:
if (!file.exists()) {
    writeBackToClient...
}{
   return file content
}